### PR TITLE
Add Docs for Device Grouping

### PR DESCRIPTION
**Pull request by jme783**
- Create a guide for device grouping
- Update the "rollout firmware" docs to more accurately reflect up-to-date product behavior

### CODE REVIEW

**Comment by monkbroc** [comment]
I suggest moving the disclaimer that this is a beta only available to paid customers to a box like on the [Raspberry Pi beta page](https://docs.particle.io/guide/getting-started/start/raspberry-pi/#particle-pi-beta).

This will avoid someone trying to follow the steps and not understanding why the buttons are disabled.

File: src/content/guide/how-to-build-a-product/device-groups.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+given device is determined by *precedence rules*, which you can read about in
+the next section.
+
+### Firmware Precedence Rules
+
+Devices in your fleet will be targeted to
+receive a version of product firmware according to these precedence
+rules:
+
+- A device that has been **individually locked** to a version of product
```
Comment (monkbroc) [inline src/content/guide/how-to-build-a-product/device-groups.md]: One more rule with higher precedence: a development device never receives automatic updates of product firmware.

File: src/content/guide/how-to-build-a-product/device-groups.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+- **Prerelease Group**: Target this subset of devices to receive
+"bleeding-edge" firmwares for early testing. This can help you get user
+feedback and/or identify potential firmware bugs before releasing to the rest of the fleet.
+- **Group by Geographic Region**: Devices that are deployed in various
+parts of the world may need to be addressed differently. Segment device
+behavior, or run different versions of firmware.
+- **Group by Permissions**: You may have distributors, service partners,
+or customers that need access to certain device groups.
+
+_Quick tip_: If you prefer, you can also [call the Particle API](!insert
```
Comment (monkbroc) [inline src/content/guide/how-to-build-a-product/device-groups.md]: Don't forget to insert link here.

File: src/content/guide/how-to-build-a-product/device-groups.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+src="/assets/images/device-groups/batch-edit-groups.png"/>
+<p class="caption">Edit groups of many devices at once
+using the Console</p>
+
+You will then be presented with a modal that will allow you to batch
+update groups of the selected devices. **Note that editing groups for a
+device may result in the device being targeted for an OTA firmware
+update**, due to product firmware releases by group.
+
+_Quick tip_: If you prefer, you can also [call the Particle API](!insert
```
Comment (monkbroc) [inline src/content/guide/how-to-build-a-product/device-groups.md]: Insert link

File: src/content/guide/how-to-build-a-product/device-groups.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+group link) directly to tag a device to a group.
+
+_Quick tip_: After tagging devices with groups, you can use the Console
+or the [Particle API](!insert link) to filter product devices by group.
```
Comment (monkbroc) [inline src/content/guide/how-to-build-a-product/device-groups.md]: Insert link

**Review by monkbroc — APPROVED** [review]
Very well written. Nice work Jeff! I added a few suggestions to do before this is merged.

**Comment by jme783** [comment]
@monkbroc thank you SO much for doing a review here. Really great to get a separate pair of eyes on this :)

**Pull request merged**